beacon_node/beacon_chain/src/migrate.rs

```diff
@@ -708,11 +708,14 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> BackgroundMigrator<E, Ho
                     StoreOp::DeleteSyncCommitteeBranch(block_root),
                 ]
             })
-            .chain(
-                abandoned_states
-                    .into_iter()
-                    .map(|(slot, state_hash)| StoreOp::DeleteState(state_hash.into(), Some(slot))),
-            )
+            .chain(abandoned_states.into_iter().flat_map(|(slot, state_hash)| {
+                [StoreOp::DeleteState {
+                    state_root: state_hash.into(),
+                    state_slot: Some(slot),
+                    // Abandoned forks will never be used, so we can safely delete the diffs
+                    prune_hot_diff: true,
```

Comment:
We can probably remove this field, as we want to keep state summaries when retaining diffs prior to finalization

Reply:
Fixed with 385f55b1d